src/compiler/Stage/Parse/Parser.elm

```diff
@@ -410,23 +410,74 @@ literalInt =
         |> P.inContext InLiteralInt
 
 
-{-| TODO escapes
-TODO Unicode escapes
--}
+-- for literalChar and, in the future, literalString
+stringHelp = 
+    P.oneOf
+    [ P.succeed (identity)
+        |. P.token (P.Token "\\" ExpectingEscapeBackslash)
+        |= P.oneOf
+            [ P.map (\_ -> '\"') (P.token (P.Token "\"" ExpectingEscapeCharacter))
+            , P.map (\_ -> '\'') (P.token (P.Token "'"  ExpectingEscapeCharacter))
+            , P.map (\_ -> '\n') (P.token (P.Token "n"  ExpectingEscapeCharacter))
+            , P.map (\_ -> '\t') (P.token (P.Token "t"  ExpectingEscapeCharacter))
+            , P.map (\_ -> '\r') (P.token (P.Token "r"  ExpectingEscapeCharacter))
+            , P.succeed identity
+                |. P.token (P.Token "u" ExpectingEscapeCharacter)
+                |. P.token (P.Token "{" ExpectingUnicodeEscapeLeftBrace)
+                |= unicode
+                |. P.token (P.Token "}" ExpectingUnicodeEscapeRightBrace)
+            ]
+    ,  P.succeed identity
+        |= P.getChompedString (P.chompIf (always True) ExpectingChar)
+        |> P.andThen (\string ->
+                    string
+                        |> String.uncons
+                        |> Maybe.map (Tuple.first >> P.succeed)
+                        |> Maybe.withDefault (P.problem (CompilerBug "Multiple characters chomped in `literalChar`"))
+                )
+    ]
+
+
 literalChar : Parser_ Literal
 literalChar =
     (P.succeed identity
         |. P.symbol (P.Token "'" ExpectingSingleQuote)
-        |= P.getChompedString (P.chompIf (always True) ExpectingChar)
+        |= stringHelp
         |. P.symbol (P.Token "'" ExpectingSingleQuote)
     )
-        |> P.andThen
-            (\string ->
-                string
-                    |> String.uncons
-                    |> Maybe.map (Tuple.first >> Char >> P.succeed)
-                    |> Maybe.withDefault (P.problem (CompilerBug "Multiple characters chomped in `literalChar`"))
-            )
+    |> P.map (\n -> Char n)
+
+unicode : Parser_ Char
+unicode =
+  P.getChompedString (P.chompWhile Char.isHexDigit)
+    |> P.andThen codeToChar
+
+
+codeToChar : String -> Parser_ Char
+codeToChar str =
+  let
+    length = String.length str 
+    code = String.foldl addHex 0 str
+  in
+  if length < 4 || length > 6 then
+    P.problem InvalidUnicodeCodePoint
+  else if 0 <= code && code <= 0x10FFFF then
+    P.succeed (Char.fromCode code)
+  else
+    P.problem InvalidUnicodeCodePoint
+
+
+addHex : Char -> Int -> Int
+addHex char total =
```

Comment:
let's use Hex.fromString instead of this. Also, it is buggy wrt. characters outside the hex range I think.

Reply:
Fixed by https://github.com/elm-in-elm/compiler/pull/20/commits/09fa46c23d95826016cbe894793732729001da4b